### PR TITLE
Clean up capitalization, parentheses, and phrasing in docs

### PR DESCRIPTION
**Pull request by jdanford**
I updated library documentation links (https://microsoft.github.io/typespec/libraries => https://typespec.io/docs/libraries), should I update the other https://microsoft.github.io links as well?

### CODE REVIEW

**Comment by jdanford** [comment]
@microsoft-github-policy-service agree

**Review by timotheeguerin — APPROVED** [review]
Thanks a lot for the fixes, we just migrated to a new changelog system unfortunately so the rush change files aren't going to work. Would you mind updating again main and running `npx changesets` (might need to run `pnpm install` first)

File: common/changes/@typespec/bundler/clean-up-docs_2024-01-25-16-59.json
Comment (timotheeguerin) [inline common/changes/@typespec/bundler/clean-up-docs_2024-01-25-16-59.json:1]: Could you delete all those files in the `common/changes` folder too now

**Comment by timotheeguerin** [comment]
Thanks [jdanford](https://github.com/jdanford) for the contribution sorry on the bad timing for the infra change

**Comment by jdanford** [comment]
No worries!

**Pull request merged**